pkg/forwarder/backoff.go

```diff
@@ -0,0 +1,54 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package forwarder
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+const (
+	// This controls the overlap between consecutive retry interval ranges. When
+	// set to `2`, there is a guarantee that there will be no overlap. The overlap
+	// will asymptotically approach 50% the higher the value is set.
+	minBackoffFactor = 2
```

Comment:
We should make the back-off behavior configurable since some customers use a proxy (which can have different behavior).

I would add `minBackoffFactor`, `maxBackoffTime`, `baseBackoffTime` and `recoveryInterval` to the configutation. I feel that may be to much. Do you think they are all needed ?

Also should we add on option to completely unblock an endpoint when a transaction is successful. I can imagine some customers asking for it when using a custom proxy.

@ofek what do you think ?